api/api_ethereum.go

```diff
@@ -737,6 +781,58 @@ func newEthRPCPendingTransaction(tx *types.Transaction) *EthRPCTransaction {
 	return newEthRPCTransaction(tx, common.Hash{}, 0, 0)
 }
 
+// formatTxToEthTxJSON formats types.Transaction to ethTxJSON.
+// Use this function for only Ethereum typed transaction.
+func formatTxToEthTxJSON(tx *types.Transaction) *ethTxJSON {
+	var enc ethTxJSON
+
+	enc.Type = hexutil.Uint64(byte(tx.Type()))
+	// If tx is not Ethereum transaction, the type is converted to TxTypeLegacyTransaction.
+	if !tx.IsEthereumTransaction() {
+		enc.Type = hexutil.Uint64(types.TxTypeLegacyTransaction)
+	}
+	enc.Hash = tx.Hash()
+	signature := tx.GetTxInternalData().RawSignatureValues()[0]
+	// Initialize signature values when it is nil.
+	if signature.V == nil {
+		signature.V = new(big.Int)
+	}
+	if signature.R == nil {
+		signature.R = new(big.Int)
+	}
+	if signature.S == nil {
+		signature.S = new(big.Int)
+	}
+	nonce := tx.Nonce()
+	gas := tx.Gas()
+	enc.Nonce = (*hexutil.Uint64)(&nonce)
+	enc.Gas = (*hexutil.Uint64)(&gas)
+	enc.Value = (*hexutil.Big)(tx.Value())
+	data := tx.Data()
+	enc.Data = (*hexutil.Bytes)(&data)
+	enc.To = tx.To()
+	enc.V = (*hexutil.Big)(signature.V)
+	enc.R = (*hexutil.Big)(signature.R)
+	enc.S = (*hexutil.Big)(signature.S)
+
+	switch tx.Type() {
+	case types.TxTypeAccessList:
+		al := tx.AccessList()
+		enc.AccessList = &al
+		enc.ChainID = (*hexutil.Big)(tx.ChainId())
+		enc.GasPrice = (*hexutil.Big)(tx.GasPrice())
+	case types.TxTypeDynamicFee:
+		al := tx.AccessList()
+		enc.AccessList = &al
+		enc.ChainID = (*hexutil.Big)(tx.ChainId())
+		enc.MaxFeePerGas = (*hexutil.Big)(tx.GasFeeCap())
+		enc.MaxPriorityFeePerGas = (*hexutil.Big)(tx.GasTipCap())
```

Comment:
Is it okay that this type doesn't set `enc.GasPrice`?

Reply:
Yeap. This works as same with Geth.
This is a response from  Geth.


```json
{
  "jsonrpc": "2.0",
  "id": 1,
  "result": {
    "raw": "0x02f86f8302edaf048505d21dba008505d21dba00829999948c9f4468ae04fb3d79c80f6eacf0e4e1dd21deee0180c001a0b6e16781d247b505aa8eaf363a6ea61e7c23c77ad64846ffda28e73e31304884a05a296904cd009506cb5ed006af016d2c5dab255966af9ba6f8fc352f75e9b079",
    "tx": {
      "type": "0x2",
      "nonce": "0x4",
      "gasPrice": null,
      "maxPriorityFeePerGas": "0x5d21dba00",
      "maxFeePerGas": "0x5d21dba00",
      "gas": "0x9999",
      "value": "0x1",
      "input": "0x",
      "v": "0x1",
      "r": "0xb6e16781d247b505aa8eaf363a6ea61e7c23c77ad64846ffda28e73e31304884",
      "s": "0x5a296904cd009506cb5ed006af016d2c5dab255966af9ba6f8fc352f75e9b079",
      "to": "0x8c9f4468ae04fb3d79c80f6eacf0e4e1dd21deee",
      "chainId": "0x2edaf",
      "accessList": [],
      "hash": "0x9cfbea4942b334050660ec7207f3323ee13e3196b06279f922404384acbf7b47"
    }
  }
}
```